php/class-migrate.php

```diff
@@ -0,0 +1,205 @@
+			'update_post_meta_cache' => false,
+			'update_post_term_cache' => false,
+			'post_status' => array_keys( get_post_stati() ),
+			'posts_per_page' => $limit,
+			'fields' => 'ids', // We will use get_post() to fetch each posts.
+		);
+
+		if ( - 1 === $limit ) {
+			$arg['no_found_rows'] = true;
+		}
+
+		$query->query( $arg );
+		if ( $dry_run ) {
+			return $query->posts;
+		}
+
+		if ( ! empty( $query->posts ) ) {
+			$has_kses = ( false !== has_filter( 'content_save_pre', 'wp_filter_post_kses' ) );
+			if ( $has_kses ) {
+				kses_remove_filters(); // Prevent KSES from corrupting JSON in post_content.
+			}
+			if ( ! class_exists( '\WP_Customize_Manager' ) ) {
+				require_once( ABSPATH . WPINC . '/class-wp-customize-manager.php' );
+			}
+			foreach ( $query->posts as $id ) {
+				$this->migrate_post( $id );
+			}
+			if ( $has_kses ) {
+				kses_init_filters();
+			}
+		}
+		if ( - 1 === $limit ) {
+			update_option( self::KEY, 1 );
+			return count( $query->posts );
+		} else {
+			return $query->found_posts;
+		}
+	}
+
+	/**
+	 * Migrate a post.
+	 *
+	 * @param int $id Post ID.
+	 *
+	 * @return int|\WP_Error maybe updated.
+	 */
+	public function migrate_post( $id ) {
+		$post = get_post( $id );
+
+		// Get data.
+		$data = json_decode( $post->post_content, true );
+		if ( json_last_error() || ! is_array( $data ) ) {
+			$data = array();
+		}
+
+		// Validate data.
+		$manager = new \WP_Customize_Manager();
+		foreach ( $data as $setting_id => $setting_params ) {
+			// Amend post values with any supplied data.
+			if ( array_key_exists( 'value', $setting_params ) ) {
+				$manager->set_post_value( $setting_id, $setting_params['value'] ); // Add to post values so that they can be validated and sanitized.
+			}
+		}
+		do_action( 'customize_register', $manager );
+
+		// Note that in addition to post data, this will include any stashed theme mods.
+		$post_values = $manager->unsanitized_post_values( array(
+			'exclude_changeset' => true,
+			'exclude_post_data' => false,
+		) );
+
+		// Update data as new changeset.
+		$manager->add_dynamic_settings( array_keys( $post_values ) );
+		$theme = get_post_meta( $id, '_snapshot_theme', true );
+		$post_data = array();
+		foreach ( $post_values as $setting_id => $setting_value ) {
+			$setting = $manager->get_setting( $setting_id );
+
+			if ( $setting && 'theme_mod' === $setting->type ) {
+				$prefixed_setting_id = $theme . '::' . $setting->id;
+			} else {
+				$prefixed_setting_id = $setting_id;
+			}
+			$post_data[ $prefixed_setting_id ] = array(
+				'value' => $setting_value,
+				'user_id' => $post->post_author,
+			);
+			if ( $setting instanceof \WP_Customize_Setting ) {
+				$post_data[ $prefixed_setting_id ]['type'] = $setting->type;
+			}
+		}
+		$maybe_updated = wp_update_post( wp_slash( array(
+			'ID' => $post->ID,
+			'post_type' => 'customize_changeset',
+			'post_content' => Customize_Snapshot_Manager::encode_json( $post_data ),
+		) ), true );
```

Comment:
Great, and since `revisions` support is added, this should result in a new entry being added.